pkg/types/hashmap/iterator.go

```diff
@@ -0,0 +1,137 @@
+package hashmap
+
+import "github.com/zeebo/xxh3"
+
+// Iter provides a map Iter to allow traversal
+type Iter struct {
+	key         Key
+	val         Val
+	m           *Map
+	buckets     []bucket
+	bucketPtr   *bucket
+	startBucket int
+	offset      uint8
+	wrapped     bool
+	i           uint8
+	bucket      int
+	checkBucket int
+}
+
+// Key returns the key at the current iterator position
+func (it *Iter) Key() Key {
+	return it.key
+}
+
+// Val returns the value / valent at the current iterator position
+func (it *Iter) Val() Val {
+	return it.val
+}
+
+// Next updates the iterator to the next element (returning false if none exists)
+func (it *Iter) Next() bool {
```

Comment:
This function would benefit from a few comments. Or a link to a Tech Talk 😏

Reply:
Well, the original function in the Go standard library is an internal function without _any_ comment, so I guess I did quite alright here by comparison. :rofl: